modules/post/windows/manage/reflective_dll_inject.rb

```diff
@@ -6,45 +6,167 @@
 require 'msf/core/post/windows/reflective_dll_injection'
 
 class MetasploitModule < Msf::Post
-  include Msf::Post::Windows::ReflectiveDLLInjection
+  Rank = NormalRanking
 
-  def initialize(info={})
-    super( update_info( info,
-      'Name'          => 'Windows Manage Reflective DLL Injection Module',
-      'Description'   => %q{
-        This module will inject into the memory of a process a specified Reflective DLL.
-      },
-      'License'      => MSF_LICENSE,
-      'Author'       => [ 'Ben Campbell'],
-      'Platform'     => [ 'win' ],
-      'SessionTypes' => [ 'meterpreter' ],
-      'References'   =>
-        [
-          [ 'URL', 'https://github.com/stephenfewer/ReflectiveDLLInjection' ]
-        ]
-    ))
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::Process
+  include Msf::Post::Windows::ReflectiveDLLInjection
 
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name' => 'Windows Manage Reflective DLL Injection
+                      Module with arguments',
+                      'Description' => '
+                      This module will inject a specified reflective DLL
+                      into the memory of a process, new or existing,
+                      passing any arguments.
+                      ',
+                      'License' => MSF_LICENSE,
+                      'Author' => 'Ben Campbell,b4rtik',
+                      'Platform' => 'win',
+                      'SessionTypes' => ['meterpreter'],
+                      'References'   =>
+                        [
+                          [ 'URL', 'https://github.com/stephenfewer/ReflectiveDLLInjection' ]
+                        ]))
     register_options(
       [
-        OptPath.new('PATH',[true, 'Reflective DLL to inject into memory of a process.']),
-        OptInt.new('PID',[true, 'Process Identifier to inject of process to inject payload.']),
-      ])
+        OptPath.new('PATH', [true, 'Reflective DLL to inject into memory of a process.']),
+        OptInt.new('PID', [false, 'Pid  to inject', 0]),
+        OptString.new('PROCESS', [false, 'Process to spawn','notepad.exe']),
+        OptString.new('ARGUMENTS', [false, 'Command line arguments']),
+        OptInt.new('WAIT', [false, 'Time in seconds to wait.', 0])
+      ], self.class
+    )
   end
 
-  # Run Method for when run command is issued
   def run
-    # syinfo is only on meterpreter sessions
+    dll_path = gen_dll_path
+    if File.file?(dll_path)
+      client.response_timeout=20
```

Comment:
I'm not sure why this needs to be set, but if you do need to set it, then when the module is done you need to restore the original value.

Reply:
I agree, I will test to be sure but I think this has not been cleaned